test/integration/targets/sqitch/tasks/main.yml

```diff
@@ -0,0 +1,76 @@
+---
+- block:
+  - name: Create dir for resources
+    file: path=/tmp/sqitch_test state=directory mode=0755
+  
+  - name: Copy resources to tmp dir
+    copy: src=files/ dest=/tmp/sqitch_test
+  
+  - name: Create DB for tests
+    postgresql_db: name=anstest encoding='UTF-8' state=present template=template0
+  
+  - name: Install sqitch
+    package: name=sqitch state=present
+    when: ansible_os_family == "Debian"
+  
+  - name: Install CPAN
+    package: name={{ item }} state=present
+    with_items:
+      - perl-devel
+      - perl-CPAN
+      - perl-DBD-Pg
+    when: ansible_os_family == "RedHat"
+  
+  - name: Install cpanminus
+    shell: "curl -L http://cpanmin.us | perl - --sudo App::cpanminus"
+    when: ansible_distribution == "CentOS" and ansible_distribution_major_version == '6'
+  
+  - name: Install cpanminus
+    package: name=cpanminus state=present
+    when: ansible_os_family == "RedHat" and not (ansible_distribution == "CentOS" and ansible_distribution_major_version == '6')
+  
+  - name: Install sqitch
+    cpanm: name=App::Sqitch notest=true
+    when: ansible_os_family == "RedHat"
+  
+  - name: Deploy in check mode with cwd
+    sqitch: command=deploy cwd=/tmp/sqitch_test plan_file=sqitch.plan target=anstest verify=true to_change=types
+    check_mode: yes
+
+  - name: Deploy in check mode without cwd
+    sqitch: command=deploy plan_file=sqitch.plan target=anstest verify=true to_change=types
+    check_mode: yes
+
+  - name: Deploy to change
+    sqitch: command=deploy cwd=/tmp/sqitch_test plan_file=sqitch.plan target=anstest verify=true to_change=types
+  
+  - name: Rebase
+    sqitch:
+      command: rebase
+      cwd: "/tmp/sqitch_test"
+      plan_file: sqitch.plan
+      target: anstest
+      verify: true
+  
+  - name: Revert to change
+    sqitch: command=revert cwd=/tmp/sqitch_test plan_file=sqitch.plan target=anstest to_change=schemas
+  
+  - name: Deploy whole plan
+    sqitch: command=deploy cwd=/tmp/sqitch_test plan_file=sqitch.plan target=anstest
+  
+  - name: Verify
+    sqitch: command=verify cwd=/tmp/sqitch_test plan_file=sqitch.plan target=anstest from_change=types to_change=tests
+  
+  - name: Check status
+    sqitch: command=status cwd=/tmp/sqitch_test plan_file=sqitch.plan target=anstest
+  
+  - name: Revert whole plan to change
+    sqitch: command=revert cwd=/tmp/sqitch_test plan_file=sqitch.plan target=anstest to_change=schemas
+  
+  - name: Destroy DB
+    postgresql_db: name=anstest state=absent
+  
+  - name: Remove resources
+    file: path=/tmp/sqitch_test state=absent
+  when: (not (ansible_distribution == "Ubuntu" and ansible_distribution_version == "14.04")) and
+        (not (ansible_os_family == "Suse"))
```

Comment:
It would be easier to understand the tests if the requirements installation was done in a separate file from the actual test execution. You could have a simple block in the main.yml which handles the distribution and version check. That block could then include a setup.yml and test.yml for the setup and execution of the tests.